### PR TITLE
Expose syscall bindings in the public API

### PR DESCRIPTION
**Pull request by CraigFe**
This is a step towards: https://github.com/mirage/irmin/issues/1009.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 - Added `filter`, removing bindings depending on a predicate (#165)
 
+- Added `Irmin_unix.Syscalls`, a module exposing various Unix bindings for
```
Comment (samoht) [inline CHANGES.md:7]: Index_unix I guess?

Comment (CraigFe) [reply CHANGES.md]: Ah, indeed. Fixed.

**Comment by CraigFe** [comment]
Will merge this once the CI passes, unless there are any objections.

File: src/unix/syscalls.ml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+let pread ~fd ~fd_offset ~buffer ~buffer_offset ~length =
+  pread fd fd_offset buffer buffer_offset length
+
+external pwrite : Unix.file_descr -> int64 -> bytes -> int -> int -> int
+  = "caml_pwrite"
+
+let pwrite ~fd ~fd_offset ~buffer ~buffer_offset ~length =
+  pwrite fd fd_offset buffer buffer_offset length
+
+external fsync : Unix.file_descr -> unit = "unix_fsync"
```
Comment (samoht) [inline src/unix/syscalls.ml:13]: as we are exporting these symbols a bit more publicly now, maybe it's a good time to check that have consistent names. Why `caml_pread` but `unix_fsync`? :-)

Comment (samoht) [reply src/unix/syscalls.ml:13]: we should make sure they won't clash with symbols defined in `tezos-stdlib-unix` and `extunix`

Comment (CraigFe) [reply src/unix/syscalls.ml:13]: `tezos-stdlib-unix` doesn't have any external definitions. `extunix` has the following:
- `caml_extunix_fsync`
- `caml_extunix_pread64`
- `caml_extunix_pwrite64`

I have no particular thoughts on what sort of naming scheme we should use. Anyone want to give an opinion?

Comment (samoht) [reply src/unix/syscalls.ml:13]: use `caml_index_{op}` seems the safest bet.

Comment (CraigFe) [reply src/unix/syscalls.ml]: Done.

**Pull request merged**